com.unity.ml-agents/Runtime/Timer.cs

```diff
@@ -151,12 +151,16 @@ public void Begin()
         /// <summary>
         /// Stop timing a block of code, and increment internal counts.
         /// </summary>
-        public void End()
+        public void End(bool isRecording)
         {
-            var elapsed = DateTime.Now.Ticks - m_TickStart;
-            m_TotalTicks += elapsed;
-            m_TickStart = 0;
-            m_NumCalls++;
+            if (isRecording)
```

Comment:
This was a bit hacky - we don't want Agents to send their rewards to the timers when we're shutting down (since a bunch will have approx 0 reward).